pkg/middleware/subpath_redirect_test.go

```diff
@@ -0,0 +1,43 @@
+package middleware
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSubPathRedirect(t *testing.T) {
```

Comment:
When testing multiple cases like this it's preffered to use [table testing](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) like this https://github.com/grafana/grafana/blob/serve-from-sub-path-redirect/pkg/api/plugin_proxy_test.go#L10